ydb/core/grpc_services/rpc_object_storage.cpp

```diff
@@ -852,8 +862,18 @@ class TObjectStorageListingRequestGrpc : public TActorBootstrapped<TObjectStorag
                     }
                     const NYdb::TPgValue pgValue{cell.IsNull() ? NYdb::TPgValue::VK_NULL : NYdb::TPgValue::VK_TEXT, pgResult.Str, getPgTypeFromColMeta(colMeta)};
                     vb.Pg(pgValue);
-                }
-                else {
+                } else if (colMeta.PType.GetTypeId() == NScheme::NTypeIds::Decimal) {
+                    using namespace NYql::NDecimal;
+
+                    const auto loHi = cell.AsValue<std::pair<ui64, i64>>();
+                    Ydb::Value valueProto;
+                    valueProto.set_low_128(loHi.first);
+                    valueProto.set_high_128(loHi.second);
+                    const NYdb::TDecimalValue decimal(valueProto, 
+                        {static_cast<ui8>(colMeta.PType.GetDecimalType().GetPrecision()), static_cast<ui8>(colMeta.PType.GetDecimalType().GetScale())});
+                    vb.Decimal(decimal);
+                    return;
```

Comment:
`return` doesn't look correct, can you explain it for me?

Reply:
https://github.com/ydb-platform/ydb/commit/0b2b82fab577a42e687f3e1d0b7ea3dbd5e26c32

Reply:
A test has been written
7250f8fb2a07641008eab18f5b84caf03bfa9d58